cognite/client/data_classes/ai.py

```diff
@@ -0,0 +1,179 @@
+        return hash((self.page_number, self.left, self.right, self.top, self.bottom))
+
+
+@dataclass
+class AnswerReference:
+    """
+    A single reference object.
+
+    The reference object specifies which file an answer is based on, and
+    has a list of locations pointing to the specific pages and bounding boxes
+    where the answer was found.
+
+    Args:
+        file_id (int): The internal id of the document
+        external_id (str | None): The external id of the document
+        instance_id (NodeId | None): The instance id of the document
+        file_name (str): The name of the document
+        locations (list[AnswerLocation]): A list of locations within the document, where the answer was found
+    """
+
+    file_id: int
+    external_id: str | None
+    instance_id: NodeId | None
+    file_name: str
+    locations: list[AnswerLocation]
+
+    @classmethod
+    def load(cls, data: dict[str, Any]) -> AnswerReference:
+        return AnswerReference(
+            file_id=data["fileId"],
+            external_id=data.get("externalId"),
+            instance_id=NodeId.load(data["instanceId"]) if "instanceId" in data else None,
+            file_name=data["fileName"],
+            locations=[AnswerLocation.load(d) for d in data.get("locations", [])],
+        )
+
+    def __hash__(self) -> int:
+        return hash((self.file_id, self.external_id, self.instance_id, self.file_name, tuple(self.locations)))
+
+
+@dataclass
+class AnswerContent:
+    """
+    A single content object.
+
+    It consists of part of the answer from the LLM along with references to
+    the documents containing the source material for the answer.
+
+    Args:
+        text (str): The extracted plain text
+        content (list[AnswerReference]): The list of references
+    """
+
+    text: str
+    references: list[AnswerReference]
+
+    @classmethod
+    def load(cls, data: dict[str, Any]) -> AnswerContent:
+        return AnswerContent(
+            text=data["text"],
+            references=[AnswerReference.load(d) for d in data.get("references", [])],
+        )
+
+
+@dataclass
+class Answer:
+    """
+    An answer returned from the Document Question Answering API.
+
+    The answer contains of multiple content objects. Each content object
+    consists of a chunk of text along with a set of references.
+
+    Each reference contains information about which document this part
+    of the answer was found in, and gives the bounding box and page number
+    of the piece of text the answer was constructed from.
+
+    Args:
+        content (list[AnswerContent]): The list of content objects.
+    """
+
+    content: list[AnswerContent]
+
+    def get_full_answer_text(self) -> str:
+        """
+        Get the full answer text. This is the concatenation of the texts from
+        all the content objects.
+        """
+        return "".join([content.text for content in self.content])
+
+    def get_all_references(self) -> list[AnswerReference]:
```

Comment:
Consider using a property (or cached_property)
```suggestion
    @property
    def all_references(self) -> list[AnswerReference]:
```